src/app/add-item/add-item.component.ts

```diff
@@ -1,68 +1,94 @@
+import { Project, ProjectsService } from './../services/projects/projects.service';
 import { TasksService, AddActionItem } from './../services/tasksService/tasks.service';
-import { Project, ProjectsService } from '../services/projects/projects.service';
-import { Component, OnInit } from '@angular/core';
-import { MatDialogRef } from '@angular/material/dialog';
+import { Component, OnInit, Inject } from '@angular/core';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
 import { Validators, FormGroup, FormBuilder } from '@angular/forms';
 import { first } from 'rxjs/operators';
 
+enum dialogMode {
+  create = 'Create',
+  edit = 'Edit'
+}
+
 @Component({
   selector: 'app-add-item',
   templateUrl: './add-item.component.html',
   styleUrls: ['./add-item.component.css']
 })
-export class AddItemComponent implements OnInit {
+export class AddOrUpdateActionItemComponent implements OnInit {
   dialogForm: FormGroup;
-  title = 'Create Action Item';
-  projects: Project[];
-  isCreatingActionItem = false;
+  dialogMode: string;
+  dialogActionButton: string;
+  projects?: Project[];
+  isSavingDialogData: boolean = false;
+  loaderVisible: boolean = true;
 
   constructor(
-    public dialogRef: MatDialogRef<AddItemComponent>,
+    public dialogRef: MatDialogRef<AddOrUpdateActionItemComponent>,
     private formBuilder: FormBuilder,
     private dialogProjects: ProjectsService,
-    private tasksService: TasksService
+    private tasksService: TasksService,
+    @Inject(MAT_DIALOG_DATA) public data: any
   ) {}
+
   ngOnInit() {
+    !this.data.item ? (this.dialogMode = dialogMode.create) : (this.dialogMode = dialogMode.edit);
```

Comment:
**CONCERN:** Why do we need this conditional statement and can't just assign the value?